src/app/components/shared/input/range/range.component.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  input,
+  output,
+  signal,
+} from "@angular/core";
+import { ControlValueAccessor, FormsModule } from "@angular/forms";
+import { toNumber } from "@helpers/typing/toNumber";
+
+@Component({
+  selector: "baw-range",
+  templateUrl: "./range.component.html",
+  styleUrl: "./range.component.scss",
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [FormsModule],
+})
+export class RangeComponent implements ControlValueAccessor {
+  public label = input("");
+  public min = input(0);
+  public max = input(100);
+  public step = input(5);
```

Comment:
it just occurred to me: the angular compiler is doing magic here to infer the attribute names, right?

Reply:
I think you might be right

https://github.com/search?q=repo%3Aangular%2Fangular%20SignalBased&type=code